torchtext/datasets/text_classification.py

```diff
@@ -373,6 +408,122 @@ def AmazonReviewFull(*args, **kwargs):
     return _setup_datasets(*(("AmazonReviewFull",) + args), **kwargs)
 
 
+def _generate_imdb_data(key, extracted_files, vocab, tokenizer, removed_tokens):
+    _data = []
+    for fname in extracted_files:
+        if 'urls' in fname:
+            continue
+        elif key in fname:
+            if 'pos' in fname:
+                txt_iter = iter(tokenizer(row) for row in io.open(fname,
+                                                                  encoding="utf8"))
+                _iter = numericalize_tokens_from_iterator(vocab,
+                                                          txt_iter,
+                                                          removed_tokens)
+                for tokens in _iter:
+                    _data.append((1,
+                                  torch.tensor([token_id for token_id in tokens]).long()))
+            elif 'neg' in fname:
+                txt_iter = iter(tokenizer(row) for row in io.open(fname,
+                                                                  encoding="utf8"))
+                _iter = numericalize_tokens_from_iterator(vocab,
+                                                          txt_iter,
+                                                          removed_tokens)
+                for tokens in _iter:
+                    _data.append((0,
+                                  torch.tensor([token_id for token_id in tokens]).long()))
+    return _data
+
+
+def IMDB(tokenizer=get_tokenizer("basic_english"),
```

Comment:
Using a function call in a function header usually raises a flake8 error since all the call to the function will use the same object reference. The function is [here](https://github.com/pytorch/text/blob/master/torchtext/data/utils.py#L74), but it's not clear to me that there won't be issue with that here.

Reply:
Indeed the typical convention is to use `None` as default and then assign a value.

Reply:
I switch to the same API used by other text classification datasets.